### PR TITLE
Fix Avro decimal defaults

### PR DESCRIPTION
**Pull request by koxudaxi**
## Summary by CodeRabbit

- **New Features**
  - Avro schema generation now supports Decimal logical-type defaults, including bytes and fixed values.
  - Generated models preserve decimal scale and trailing zeros across supported output formats.
  - Decimal defaults are handled consistently in unions, nullable fields, collections, nested records, and referenced types.
  - Invalid precision or scale configurations now produce clear errors.

- **Documentation**
  - Added guidance describing Decimal default decoding, scale preservation, and validation limits.

- **Tests**
  - Expanded coverage across supported model types, output formats, deserialization modes, and edge cases.

### CODE REVIEW

File: tests/data/expected/main/avro/decimal_defaults_dataclass.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        default_factory=lambda: {'z': Decimal_aliased('0E-2'), 'a': Decimal_aliased('1200E-2')}
+    )
+    union: Decimal_aliased | None = Decimal_aliased('1E-2')
+    string_first: str | Decimal_aliased | None = '12'
+    nullable: Decimal_aliased | None = None
+    record: Ledger | None = field(
+        default_factory=lambda: Ledger(amount=Decimal_aliased('258E-2'))
+    )
+    record_array: list[Ledger] | None = field(
+        default_factory=lambda: [{'amount': Decimal_aliased('-258E-2')}]
```
[the remark this answers is not in the document]
Comment (koxudaxi) [reply tests/data/expected/main/avro/decimal_defaults_dataclass.py:47]: Confirmed the same existing behavior with ordinary non-logical Avro record arrays on the parent revision: standard dataclass defaults retain dictionary elements. Decimal leaf conversion preserves that container behavior and changes only the diagnosed decimal representation. A generator-level recursive dataclass construction change needs separate coverage across input formats; editing this golden alone would assert behavior that the generator does not implement. This limitation is recorded separately from this decimal-default PR.

File: src/datamodel_code_generator/parser/avro.py [hunk 2]
```diff
@@ -505,6 +514,31 @@ def _convert_default_type(self, value: Any, schema: Any, namespace: str | None)
...
+            or precision == 0
+        ):
+            return value
+
+        from decimal import MIN_ETINY, Decimal  # ruff: ignore[import-outside-top-level]
+
+        if scale > -MIN_ETINY:
+            msg = f"Avro decimal default scale is outside the Python Decimal range: {scale}"
+            raise Error(msg)
+        coefficient = Decimal(int.from_bytes(value, "big", signed=True))
```
[the remark this answers is not in the document]
Comment (koxudaxi) [reply src/datamodel_code_generator/parser/avro.py:541]: Fixed in 98f55a36. Fixed decimal defaults now reject a decoded byte count that differs from the declared fixed size before constructing Decimal. External invalid fixtures exercise both CLI and API; the changed validation paths have 100% line/branch coverage. The final Avro integration passes 311 tests (one existing skip), and valid default output bytes/order remain unchanged.

[the remark this answers is not in the document]
Comment (koxudaxi) [reply src/datamodel_code_generator/parser/avro.py:541]: Fixed in 98f55a36. The decoded coefficient now rejects values exceeding precision, using Decimal.adjusted() without depending on the active Decimal context. CLI/API external fixtures cover the precision=2 coefficient=300 case and valid boundary controls. The six added validation lines and four branch arcs are fully covered; valid decimal output and Decimal context flags are unchanged.
